src/test/java/com/synopsys/integration/stepworkflow/StepWorkflowTest.java

```diff
@@ -337,43 +261,11 @@ public void testConditionalFailedPreviousResponse() {
 
         // Call run() and validate return StepWorkflowResponse
         StepWorkflowResponse<Integer> response = buildStepWorkflow.run();
-        assertFalse(response.wasSuccessful());
-        assertNull(response.getData());
-        assertEquals(exception, response.getException());
-    }
-
-    @Test
-    public void testConditionalConstructorAndThen() {
-        // Create and validate then() Builder
-        StepWorkflow.Builder<Integer> thenBuilder = successfulBuilderOfSupplierOne.then(successfulSupplierTwo);
-        assertEquals(successfulSupplierOne, thenBuilder.start.step);
-        assertNotNull(thenBuilder.start.next);
-        assertEquals(successfulSupplierTwo, thenBuilder.end.step);
-
-        // Create and validate build() Builder using conditionalBuilderOne.then() and ConditionalBuilder constructor
-        StepWorkflow.ConditionalBuilder<Integer, Integer> conditionalBuilder = new StepWorkflow.ConditionalBuilder(thenBuilder, successfulSupplierThree);
-        StepWorkflow.ConditionalBuilder<Integer, Object> thenConditionalBuilder = conditionalBuilder.then(successfulFunctionOne);
-        StepWorkflow.Builder<Object> buildBuilder = thenConditionalBuilder.build(successfulFunctionTwo);
-        assertEquals(successfulSupplierOne, buildBuilder.start.step);
-        assertEquals(successfulSupplierTwo, buildBuilder.start.next.step);
-        assertEquals(successfulFunctionTwo, buildBuilder.end.step);
-
-        // Run and validate build()
-        StepWorkflow<Object> buildStepWorkflow = buildBuilder.build();
-        assertEquals(buildBuilder.start, buildStepWorkflow.start);
-        assertEquals(buildBuilder.end, buildStepWorkflow.end);
-
-        // Call run() and validate return StepWorkflowResponse
-        StepWorkflowResponse<Object> response = buildStepWorkflow.run();
         assertTrue(response.wasSuccessful());
-        assertEquals(TEST_DATA_FIVE, response.getData());
+        assertEquals(TEST_DATA_THREE, response.getData());
         assertNull(response.getException());
     }
 
-    private boolean returnValue(boolean returnValue) {
-        return returnValue;
-    }
-
     private void successfulExecutor() {
```

Comment:
All of these private methods can and likely should be converted into individual lambda expressions when the substeps are created. As they are now, it's difficult to take in exactly what the substeps used in testing are doing, since the method that is ostensibly the implementation of the substep is at the bottom of the file and the definition of the substep is at the top.